addons/account/models/account_invoice.py

```diff
@@ -901,11 +902,16 @@ def action_invoice_open(self):
     def action_invoice_paid(self):
         # lots of duplicate calls to action_invoice_paid, so we remove those already paid
         to_pay_invoices = self.filtered(lambda inv: inv.state != 'paid')
-        if to_pay_invoices.filtered(lambda inv: inv.state != 'open'):
+        if to_pay_invoices.filtered(lambda inv: inv.state not in ('open', 'in_payment')):
             raise UserError(_('Invoice must be validated in order to set it to register payment.'))
         if to_pay_invoices.filtered(lambda inv: not inv.reconciled):
             raise UserError(_('You cannot pay an invoice which is partially paid. You need to reconcile payment entries first.'))
-        return to_pay_invoices.write({'state': 'paid'})
+
+        for invoice in to_pay_invoices:
```

Comment:
Note: there is an issue with the following use case

Journal is with option post_at_bank_rec = True, we have an invoice that is fully reconciled with a draft move. With those changes, the reconciled field on the invoice is set to True but the state is still 'in_payment'. The problem come if the user change the settings on the journal and set post_at_bank_rec = False. After that he can post manually the move, but the invoice state won't change because this won't retrigger a write on it. Therefore user will be stuck with an invoice in state 'in_payment' until there is a write on that particular invoice.
One possible solution is when posting a move, checking if invoice associated is in 'in_payment' state and if yes call the action_invoice_paid method? (there might be better solution, might need to brainstorm on it)